File: backend/app/schemas/interaction.py
from pydantic import BaseModel
from uuid import UUID
from typing import Optional
from enum import Enum


class ActionType(str, Enum):
    LIKE = "like"
    DISLIKE = "dislike"
    SAVE = "save"
    SKIP = "skip"
    WATCH_COMPLETE = "watch_complete"


class InteractionCreate(BaseModel):
    clip_id: UUID
    action: ActionType
    watch_duration_ms: Optional[int] = None
    session_id: Optional[UUID] = None


class InteractionResponse(BaseModel):
    id: UUID
    clip_id: UUID
    action: str
    created_at: str

    model_config = {"from_attributes": True}

File: ios/BYETZ/Models/Interaction.swift
import Foundation

enum ActionType: String, Codable {
    case like
    case dislike
    case save
    case skip
    case watchComplete = "watch_complete"
}

struct InteractionRequest: Codable {
    let clipId: UUID
    let action: ActionType
    let watchDurationMs: Int?
    let sessionId: UUID?

    enum CodingKeys: String, CodingKey {
        case clipId = "clip_id"
        case action
        case watchDurationMs = "watch_duration_ms"
        case sessionId = "session_id"
    }
}

struct InteractionResponse: Codable {
    let id: UUID
    let clipId: UUID
    let action: String
    let createdAt: String

    enum CodingKeys: String, CodingKey {
        case id
        case clipId = "clip_id"
        case action
        case createdAt = "created_at"
    }
}
